lib/core.js

```diff
@@ -295,18 +295,26 @@ declare class JSON {
 }
 
 /* Iterators */
-interface IteratorResult<T> {
+interface IteratorResult<Y,R> {
     done: boolean;
-    value?: T;
+    value?: Y|R;
```

Comment:
The value is optional—it becomes undefined upon completion of the iterator. i.e., when `done=false`, `value=undefined`.